tinkerbell/stack/templates/nginx.yaml

```
@@ -1,6 +1,10 @@
 {{- if .Values.stack.enabled }}
 {{- $sourceInterface := .Values.stack.relay.sourceInterface -}}
-{{- $macvlanInterfaceName := printf "%s%s" "macvlan" (randNumeric 2) -}}
+{{- $dhcpInterfaceType := "macvlan" -}}
+{{- if eq .Values.stack.relay.interfaceMode "ipvlan" -}}
+{{- $dhcpInterfaceType = "ipvlan" -}}
+{{- end -}}
```

```suggestion
{{- $ifaceModes := dict "ipvlan" "" "macvlan" "" -}}
{{- if not hasKey $ifaceModes .Values.stack.relay.interfaceMode -}}
  {{- fail "invalid value in .stack.relay.interfaceMode: valid modes include ipvlan and macvlan" -}}
{{- end -}}
```
Do we want something like this so there's strong feedback for the user if they specify something invalid?